powertools-large-messages/src/main/java/software/amazon/lambda/powertools/largemessages/LargeMessageConfig.java

```diff
@@ -0,0 +1,59 @@
+package software.amazon.lambda.powertools.largemessages;
+
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.S3ClientBuilder;
+
+import static software.amazon.lambda.powertools.core.internal.LambdaConstants.AWS_REGION_ENV;
+
+/**
+ * Singleton instance for Large Message Config.
+ * <br/>
+ * Optional: Use it in your Lambda constructor to pass a custom {@link S3Client} to the {@link software.amazon.lambda.powertools.largemessages.internal.LargeMessageProcessor}
+ * <br/>
+ * If you don't use this, a default S3Client will be created.
+ * <pre>
+ * public MyLambdaHandler() {
+ *     LargeMessageConfig.init().withS3Client(S3Client.create());
+ * }
+ * </pre>
+ */
+public class LargeMessageConfig {
+
+    private static final LargeMessageConfig INSTANCE = new LargeMessageConfig();
+    private S3Client s3Client;
+
+    private LargeMessageConfig() {
+    }
+
+    public static LargeMessageConfig get() {
```

Comment:
I think we should have javadoc on every public interface in the powertools API. WDYT?